backend/src/main/java/com/mojprofesor/backend/service/ProfessorService.java

```diff
@@ -0,0 +1,86 @@
+package com.mojprofesor.backend.service;
+
+import com.mojprofesor.backend.entity.ProfessorEntity;
+import com.mojprofesor.backend.payload.AddProfessorRequest;
+import com.mojprofesor.backend.payload.ProfessorResponse;
+import com.mojprofesor.backend.repository.ProfessorRepository;
+import javassist.NotFoundException;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Service
+public class ProfessorService {
+
+    private final ProfessorRepository professorRepository;
+
+    public Page<ProfessorResponse> getAllProfessors(Integer page, Integer size) {
+        Pageable requestPage = PageRequest.of(page, size);
+
+        List<ProfessorResponse> response = professorRepository.findAll(requestPage).stream().map(
+                ProfessorResponse::of
+        ).collect(Collectors.toList());
+
+        return new PageImpl<>(response);
+    }
+
+    public ProfessorResponse getProfessorById(@NonNull Long id) throws NotFoundException {
+        Optional<ProfessorEntity> professor = professorRepository.findById(id);
+
+        if (professor.isPresent()) {
```

Comment:
możesz napisać tak

ProfessorEntity professorEntity = professor
                .orElseThrow(() -> new NotFoundException("Not found professor of id: " + id)); // jeśli istnieje to zwróci encje, a jak nie to rzuci wyjątek
        return ProfessorResponse.of(professorEntity);